featuretools/computational_backends/pandas_backend.py

```diff
@@ -217,7 +154,12 @@ def calculate_all_features(self, instance_ids, time_last,
                                    list(instance_ids)[0]), 'w') as f:
                 pstats.Stats(pr, stream=f).strip_dirs().sort_stats("cumulative", "tottime").print_stats()
 
-        df = finished_entities[self.target_eid]
+        # The dataframe for the target entity should be stored at the root of
+        # df_trie.
+        df = df_trie[[]]
+
+        if df.empty:
+            return self.generate_default_df(instance_ids=instance_ids)
```

Comment:
I've played around with this a bit and I don't think it makes sense to move. This line is necessary because `calculate_feature_matrix` is required to return a dataframe with rows for each of the given `instance_ids`. But `_calculate_features_for_entity` has no such contract, so it seems reasonable for it to generate empty dataframes. It is not given specific instances to calculate features for (only a set of values to filter by), so I don't see how it would call `generate_default_df`.

I also considered adding an early return to `_calculate_features_for_entity` if the dataframe is empty (without populating it with empty rows), but we can't do this either because `df_trie` still needs to be filled with empty dataframes. For example, if we are calculating `COUNT(customers.transactions)` but the `customers` df is empty we still need to recurse and generate an empty df for `transactions`.